src/Template/User/settings.ctp

```diff
@@ -172,21 +172,21 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Settings')));
             </md-list-item>
 
             <md-list-item>
-                <?php $hideRandomSentence = $userSettings->settings['hide_random_sentence']; ?>
+                <?php $showRandomSentence = $userSettings->settings['hide_random_sentence']; ?>
```

Comment:
With your change, the name of the variable `$showRandomSentence` doesn’t make sense any more. It should be either `$hideRandomSentence = $userSettings->settings['hide_random_sentence'];` or `$showRandomSentence = !$userSettings->settings['hide_random_sentence'];`.

Reply:
That is very true! I modified the name of the variable :)

Reply:
Yes, this is still confusing. How about:
```
$showRandomSentence = !$userSettings->settings['hide_random_sentence'];
ng-false-value="0"
ng-true-value="1"
ng-init="showRandomSentence = <?= (int)$showRandomSentence ?>"
`

Reply:
Since the value we display and the one we send are opposite, isn't it one of the following? I have a preference for the first one but Is there a clean version without the `1 -`?
```
$showRandomSentence = !$userSettings->settings['hide_random_sentence'];
ng-false-value="0"
ng-true-value="1"
ng-init="showRandomSentence = <?= (int)$showRandomSentence ?>"

'value' => '{{1 - showRandomSentence}}'
```

```
$showRandomSentence = !$userSettings->settings['hide_random_sentence'];
ng-false-value="1"
ng-true-value="0"
ng-init="showRandomSentence = <?= 1 - (int)$showRandomSentence ?>"

'value' => '{{showRandomSentence}}'
```

Reply:
Yes, I forgot about the label showing the opposite of the value. I think it’s simpler to keep hide* variable names and just switch the tick:
```
$hideRandomSentence = $userSettings->settings['hide_random_sentence'];
ng-false-value="1"
ng-true-value="0"
ng-init="hideRandomSentence = <?= (int)$hideRandomSentence ?>"

'value' => '{{hideRandomSentence}}'
```

Reply:
I agree with jiru. Renaming the model name just makes the code more complicated without any real advantage IMHO. I would rather add a comment explaining the unusual `true`/`false` values and the opposite label text.